packages/typespec-ts/src/modular/helpers/typeHelpers.ts

```diff
@@ -4,6 +4,30 @@ import {
   SdkType
 } from "@azure-tools/typespec-client-generator-core";
 
+export function getAdditionalPropertiesType(
+  type: SdkType | undefined
+): SdkType | undefined {
+  if (
+    !type ||
+    !(
+      "additionalProperties" in type ||
+      (type.kind === "model" && getAdditionalPropertiesType(type.baseModel))
+    )
+  ) {
+    return undefined;
+  }
+
+  if (type.additionalProperties) {
+    return type.additionalProperties;
+  }
+
+  if (type.baseModel) {
+    return getAdditionalPropertiesType(type.baseModel);
```

Comment:
Could you help double check what the additionalProperties type is for model B? I just wonder if tcgc or compiler has already done the union here or is it something emitter should do?
```ts
model A {
  ...Record<string>
}

model B extends A {
  ...Record<bytes>
}
```

Reply:
Our generation is correct but typespec would report errors for non-compile code, let's wait their fix and turn on this case.